src/pages/SharePage.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import {View} from 'react-native';
+import {SafeAreaProvider} from 'react-native-safe-area-context/src/SafeAreaContext';
+import {SafeAreaInsetsContext} from 'react-native-safe-area-context';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import ONYXKEYS from '../ONYXKEYS';
+import ShareManager from '../libs/ShareManager';
+import {redirect} from '../libs/actions/App';
+import {clear as clearSharedItem} from '../libs/actions/SharedItem';
+import {hide as hideSidebar} from '../libs/actions/Sidebar';
+import {addAction} from '../libs/actions/Report';
+import ROUTES from '../ROUTES';
+import CustomStatusBar from '../components/CustomStatusBar';
+import SidebarLinks from './home/sidebar/SidebarLinks';
+import FAB from '../components/FAB';
+import CreateMenu from '../components/CreateMenu';
+import * as ChatSwitcher from '../libs/actions/ChatSwitcher';
+
+const propTypes = {
+    // Currently viewed reportID
+    currentlyViewedReportID: PropTypes.string,
+
+    // Shared item object
+    sharedItem: PropTypes.shape({
+        // Shared item type (one of the `ShareType.*`)
+        type: PropTypes.string.isRequired,
+
+        // Shared item data. Differs depending on type
+        data: PropTypes.oneOfType([
+            // Text or HTML for `ShareType.TEXT`
+            PropTypes.string,
+
+            // File data structure for `ShareType.FILE`
+            PropTypes.shape({
+                // File name
+                name: PropTypes.string.isRequired,
+
+                // File MIME type
+                type: PropTypes.string.isRequired,
+
+                // File URI
+                uri: PropTypes.string.isRequired,
+            }),
+        ]).isRequired,
+    }),
+};
+
+const defaultProps = {
+    currentlyViewedReportID: '',
+    sharedItem: null,
+};
+
+class SharePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isCreateMenuActive: false,
+        };
+
+        this.onCloseButtonClick = this.onCloseButtonClick.bind(this);
+        this.onLinkClick = this.onLinkClick.bind(this);
+        this.onCreateMenuItemSelected = this.onCreateMenuItemSelected.bind(this);
+        this.toggleCreateMenu = this.toggleCreateMenu.bind(this);
+        this.addSharedItemToReport = this.addSharedItemToReport.bind(this);
+    }
+
+    /**
+     * Clears the shared item and closes Share Page.
+     */
+    onCloseButtonClick() {
+        clearSharedItem();
+        redirect(this.props.currentlyViewedReportID !== ''
+            ? ROUTES.getReportRoute(this.props.currentlyViewedReportID)
+            : ROUTES.HOME);
+    }
```

Comment:
Seems like duplicated redirect logic here. Maybe could have abstracted this somehow.